content/tutorials/reactor-firehose-kinesis.textile

```diff
@@ -0,0 +1,602 @@
+
+Enter the following code in @server.js@:
+
+```[javascript]
+// server.js
+
+require("dotenv").config()
+
+const express = require("express")
+const rn = require("random-number")
+const Ably = require("ably")
+const ably = new Ably.Rest({ key: process.env.ABLY_API_KEY })
+
+const app = express()
+
+// make all the files in 'public' available
+app.use(express.static("public"))
+
+// load the home page, index.html
+app.get("/", (request, response) => {
+  response.sendFile(__dirname + "/views/index.html")
+})
+
+// listen for requests
+const listener = app.listen(process.env.PORT, () => {
+  console.log("Your app is listening on port " + process.env.PORT)
+})
+```
+
+This code imports all the required dependencies, instantiates the "Ably REST API client":/documentation/rest and serves the @index.html@ web page and supporting files. Test it by running @node server.js@ and visiting @http://localhost:5000@ in your browser. If everything is working correctly, you should see a web page with a button inviting you to generate sensor readings. The button doesn't do anything yet: you will write the code for this in the next step. 
+
+h2. Step 5: Generate random sensor data
+
+In this step, you will use the Node.js @random-number@ module to generate random sensor readings that contain temperature and humidity values and then publish them to the @raw-sensor-data@ channel. Because you created a Reactor Firehose integration rule on that channel, the transactions you created will be sent to your data stream.
+
+First, modify your @server.js@ file to include the @/generate@ route, just below your home (@/@) route:
+
+```[javascript]
+app.get("/generate", (request, response) => {
+  for (i = 0; i < process.env.NUM_SENSORS; i++) {
+    let data = {
+      temperature: rn({ min: -10, max: 50, integer: false }), // min, max, integer?
+      humidity: rn({ min: 10, max: 99, integer: true }), // min, max, integer?
+    }
+
+    console.log(data)
+
+    channel.publish("reading", JSON.stringify(data))
+  }
+  response.sendStatus(200)
+})
+```
+
+Then, add a @channel@ variable just after you have instantiated the Ably REST API client, and use it to retrieve the @raw-sensor-data@ channel:
+
+```[javascript]
+...
+const Ably = require("ably")
+const ably = new Ably.Rest({ key: process.env.ABLY_API_KEY })
+
+const channel = ably.channels.get("raw-sensor-data")
+```
+
+Finally, create the button click event handler in @public/client.js@ that will make a request to the @/generate@ route when you click the Generate Readings button in the @index.html@ page:
+
+```[javascript]
+function generateSensorData() {
+  fetch("/generate")
+}
+```
+
+You are now ready to test your integration rule.
+
+h2. Step 6: Test your data stream
+
+Execute @node server.js@ and visit @http://localhost:5000@ in your browser.
+
+Click the Generate Readings button. Your server will generate a number of random transactions equal to the @NUM_SENSORS@ setting in @.env@ and output them to the console:
```

Comment:
Would also suggest "Generate Readings" is bolded too.

Reply:
Done, for all occurrences